File: Daily Test/Java/Sort Integers In Submatrix 7.12.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int r = sc.nextInt(), c = sc.nextInt();
        int[][] a = new int[r][c];
        for(int i = 0; i < r; i++) {
            for(int j = 0; j < c; j++) {
                a[i][j] = sc.nextInt();
            }
        }
        int x = sc.nextInt(), y = sc.nextInt();
        int m = x-1, n = y-1, w = c-y+1;
        
        ArrayList<Integer> l = new ArrayList<>();
        for(int i = 0; i < x; i++) {
            for(int j = n; j < n+w; j++) {
                l.add(a[i][j]);
            }
        }
        Collections.sort(l);
        
        int k = 0;
        for(int i = 0; i < x; i++) {
            for(int j = n; j < n+w; j++) {
                a[i][j] = l.get(k++);
            }
        }
        
        for(int[] row : a) {
            for(int val : row) {
                System.out.print(val + " ");
            }
            System.out.println();
        }
        sc.close();
    }
}

File: Daily Test/Python/Sort Integers In Submatrix 7.12.2025.py
r, c = map(int, input().split())
a = [list(map(int, input().split())) for _ in range(r)]
x, y = map(int, input().split())
m, n, w = x-1, y-1, c-y+1

l = []
for i in range(x):
    for j in range(n, n+w):
        l.append(a[i][j])
l.sort()

k = 0
for i in range(x):
    for j in range(n, n+w):
        a[i][j] = l[k]
        k += 1

for row in a:
    print(*row)